x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java

```diff
@@ -2739,6 +2814,18 @@ private void assertMsg(String logLine, Map<String, String> checkFields, Map<Stri
                 logLine = logEntryFieldPattern.matcher(logLine).replaceFirst("");
             }
         }
+        // check each string-valued literal field
+        for (final Map.Entry<String, String> checkField : checkLiteralFields.entrySet()) {
+            assertThat("Use checkFields to mark missing fields", checkField.getValue(), notNullValue());
+            final Pattern logEntryFieldPattern = Pattern.compile(Pattern.quote("\"" + checkField.getKey() + "\":" + checkField.getValue()));
+            assertThat(
+                "Field " + checkField.getKey() + " value mismatch. Expected " + checkField.getValue(),
+                logEntryFieldPattern.matcher(logLine).find(),
+                is(true)
+            );
+            // remove checked field
+            logLine = logEntryFieldPattern.matcher(logLine).replaceFirst("");
```

Comment:
The issue is that for "normal" checked fields, quotes the value:

`final String quotedValue = "\"" + checkField.getValue().replaceAll("\"", "\\\\\"") + "\"";` 

So the matcher looks for `"{...}"` instead of `{...}`. I introduced `checkLiteralFields` to be able to pass a string that does not get modified, i.e., is the literal string to match on.

Reply:
Right. I missed that. Thanks for the explanation.